cleanlab/multilabel_classification/dataset.py

```diff
@@ -0,0 +1,91 @@
+import pandas as pd
+import numpy as np
+from cleanlab.filter import _find_multilabel_issues_per_class
+from cleanlab.internal.multilabel_utils import get_onehot_num_classes
+from collections import defaultdict
+
+
+def common_multilabel_issues(
+    labels=list,
+    pred_probs=None,
+    *,
+    class_names=None,
+    confident_joint=None,
+) -> pd.DataFrame:
+    """Summarizes which classes in a multi-label dataset appear most often mislabeled overall.
+
+    This method works by providing any one (and only one) of the following inputs:
+
+    1. ``labels`` and ``pred_probs``, or
+    2. ``confident_joint``
+
+    Only provide **exactly one of the above input options**, do not provide a combination.
+
+    Parameters
+    ----------
+    labels : List[List[int]]
+        Refer to documentation for this argument in :py:func:`filter._find_multilabel_issues_per_class <cleanlab.filter._find_multilabel_issues_per_class>` for further details.
+
+    pred_probs : np.ndarray, optional
+      Refer to documentation for this argument in :py:func:`filter._find_multilabel_issues_per_class <cleanlab.filter._find_multilabel_issues_per_class>` for further details.
+
+
+    class_names : Iterable[str]
+        A list or other iterable of the string class names. The list should be in the order that
+        matches the label indices. So if class 0 is 'dog' and class 1 is 'cat', then
+        ``class_names = ['dog', 'cat']``.
+
+    confident_joint : np.ndarray, optional
+        Refer to documentation for this argument in :py:func:`filter._find_multilabel_issues_per_class <cleanlab.filter._find_multilabel_issues_per_class>` for further details.
+
+
+    Returns
+    -------
+    common_multilabel_issues : pd.DataFrame
+        DataFrame where each row corresponds to a Class (specified as the row-index) with columns "In Given Label", "In Suggested Label", "Num Examples", "Issue Probability".
```

Comment:
I'm not sure the code below actually does this?

Reply:
return (
        pd.DataFrame.from_dict(summary_issue_counts)
        .set_index("Class Index")
        .sort_values(by=["Issue Probability"], ascending=False)
    )

Reply:
We decided to leave it it to be indexed by class index, and a separate column called class that will contain class name (if provided, else class index as class name)

Reply:
Let's not set the class index or any row index for the returned DF. There is no point IMO, because we now have the "Class" column in the returned DF.

That should be the single source of truth. If the user provided class_names, then they get the strings in this column. If they didn't, then they get the integer indices in this column.  So no need for a row-index that is redundant.